lib/util/location.ex

```diff
@@ -97,6 +97,89 @@ defmodule Util.Location do
     distance(as_location!(lhs), as_location!(rhs))
   end
 
+  @doc """
+  Returns the distance between the point and the segment defined by
+  segment_start and segment_end, as well as the point on the segment
+  that provides that minimum distance.
+
+  ## Examples
+      iex> Util.Location.distance_from_segment(
+      ...>   Util.Location.new(42.00004, -71.00014),
+      ...>   {Util.Location.new(42, -71.0002), Util.Location.new(42.0001, -71.0001)}
+      ...> )
+      %{
+        closest_point: %Util.Location{latitude: 42.00004711559566, longitude: -71.00015288440434},
+        distance: 1.3264961988869075
+      }
+  """
+  @spec distance_from_segment(
```

Comment:
Bikeshed time!

Is this the right name for this function? I'm leaning towards renaming it to `nearest_point_to_segment` or something like that.

Also, is this the right file for this? It does have to do with `Location`'s, which is why I put it here, but maybe this is bespoke enough that it should get its own file?

Reply:
I like `nearest_point_to_segment`, but I think that maybe this function shouldn't return distance, it's not that hard to compute it later, so I'd prefer that instead, but I'd like to hear why this should return the distance value.